project_task_code/models/project_task.py

```diff
@@ -42,3 +42,15 @@ def name_get(self):
             name = "[{}] {}".format(rec.code, task[1])
             new_result.append((rec.id, name))
         return new_result
+
+    @api.model
+    def _name_search(
```

Comment:
**issue:** override the public method if possible (and it is).

```suggestion
    def name_search(
```

This is a good practice because the code is more readable and predictable. Public methods are documented. Thus, more developers know about them.

Besides FWIW, the private method was created [to allow overriding the user that issues the search](https://github.com/odoo/odoo/blob/8dddaa61c86d0942fc5a7487707feb6d0da0e1dc/odoo/models.py#L1913-L1914). This is not your case.